tasks/main.yml

```diff
@@ -1,47 +1,18 @@
 ---
-- name: "Ensure container cluster is {{ os_container_infra_state }}"
-  os_container_infra:
-    auth_type: "{{ os_container_infra_auth_type or omit }}"
-    auth: "{{ os_container_infra_auth or omit }}"
-    cloud: "{{ os_container_infra_cloud or omit }}"
-    state: "{{ os_container_infra_state }}"
-    cluster_name: "{{ os_container_infra_cluster_name }}"
-    cluster_template_name: "{{ os_container_infra_cluster_template_name }}"
-    master_count: "{{ os_container_infra_master_count }}"
-    node_count: "{{ os_container_infra_node_count }}"
-    keypair: "{{ os_container_infra_keypair }}"
+- name: os_container_infra_state != 'upgrade'
+  include_tasks: not-upgrade.yml
```

Comment:
Can you think of a clearer name than `not-upgrade`?

Reply:
present-or-query.yml